queue_listeners/lib/actionkit_page_parser.rb

```diff
@@ -0,0 +1,61 @@
+require_relative '../config/load_env'
+require_relative '../lib/crm_page'
+require 'json'
+
+class ActionkitPageParser
+  def initialize
+    @base_url = ENV['ACTIONKIT_BASE_URL']
+    @widget_to_page_types = {petition: 'Petition', donation: 'Donation'}
+  end
+
+  def parse_from_message(message)
```

Comment:
Yep. I just pushed an update with the individual parse types (message and AK) as private nested classes, which I think will be in line with what you were thinking. Let me know!